File: Minimum-Sum-of-Four-Digit-Number-After-Splitting-Digits.cpp
class Solution {
public:
    int minimumSum(int num) {
        vector<int> digits;
        while (num > 0) {
            digits.push_back(num % 10);
            num /= 10;
        }

        for (int i = 0; i < digits.size(); ++i) {
            for (int j = i + 1; j < digits.size(); ++j) {
                if (digits[i] > digits[j]) {
                    int temp = digits[i];
                    digits[i] = digits[j];
                    digits[j] = temp;
                }
            }
        }

        int num1 = digits[0] * 10 + digits[2];
        int num2 = digits[1] * 10 + digits[3];
        return num1 + num2;
    }
};

File: Minimum-Sum-of-Four-Digit-Number-After-Splitting-Digits.java
class Solution {
    public int minimumSum(int num) {
        int[] digits = new int[4];
        int i = 0;
        while (num > 0) {
            digits[i] = num % 10;
            num /= 10;
            i++;
        }

        for (int a = 0; a < 4; a++) {
            for (int b = a + 1; b < 4; b++) {
                if (digits[a] > digits[b]) {
                    int temp = digits[a];
                    digits[a] = digits[b];
                    digits[b] = temp;
                }
            }
        }

        int num1 = digits[0] * 10 + digits[2];
        int num2 = digits[1] * 10 + digits[3];
        return num1 + num2;
    }
}
